src/main/java/dev/jbang/Jar.java

```diff
@@ -0,0 +1,72 @@
+package dev.jbang;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * A Jar represents a RunUnit (something runnable) in the form of a JAR file.
+ * It's a reference to an already existing JAR file, either as a solitary file
+ * on the user's file system or accessible via a URL. But it can also be a Maven
+ * GAV (group:artifact:version) reference resolving to a JAR file in the Maven
+ * cache (~/.m2/repository).
+ *
+ * NB: The Jar contains/returns no other information than that which can be
+ * extracted from the JAR file itself. So all Jars that refer to the same JAR
+ * file will contain/return the exact same information.
+ */
+public class Jar implements RunUnit {
```

Comment:
JarUnit would be better here

Reply:
JarUnit and ScriptUnit?

Reply:
An other option instead of "RunUnit" would be something like "Executable" (or "Runnable" but then you get all the conflicts with the existing Java Runnable)

Reply:
Ok, I see you wrote stuff in the chat too. ScriptUnit, RunnableUnit, etc. Let's discuss that some more there first...